sdks/java/core/pom.xml

```diff
@@ -430,6 +430,19 @@
 
     <!-- test dependencies -->
     <dependency>
+      <groupId>org.apache.beam</groupId>
+      <artifactId>beam-sdks-common-fn-api</artifactId>
+      <type>test-jar</type>
```

Comment:
You seem to have pointed out 2 things:

1. `type` vs `classifier` (honestly not sure -- can you say more about why that was wrong in past?)
1. `<scope>test</scope>`

Re these questions:

1. This is a "what are we depending on" -- we want the test resources from the common API (probably not Fn, I agree).
1. This is a "when do we depend on it" -- we only need it in test scope.